services/data/src/react/hooks/useQueryExecutor.ts

```diff
@@ -11,6 +11,14 @@ interface StateType<T> {
     data?: T
 }
 
+const checkVariables = (variables: any) => {
+    if (typeof variables !== 'object' || Array.isArray(variables)) {
+        throw new Error(
+            'Query/mutation variables must be represented as a plain object.'
+        )
```

Comment:
I know we talked about this throwing an error, but I'm worried about this being quite disruptive.  If a developer passes through a variable that happens to sometimes NOT be a plain object, I don't think we should punish the user with a full (or nearest error boundary) screen of death.  I think we might want to be a bit more lenient here, at least at first, by just doing a rare `console.error` and bailing out of the fetch.  What do you think @mediremi @varl?

Reply:
You're right that throwing a runtime error would be disruptive and in fact it would make this PR a breaking change (so a minor release instead of a patch release). Let's do `console.error` for now and maybe not even bail out of the fetch?

Reply:
Side note:

>(so a minor release instead of a patch release)

Anything (`fix`, `chore`, `feat`, etc.) can be a breaking change by adding the `BREAKING CHANGE:` line in the commit msg.

Related to the question:

If we log an error, I think we should **stop** processing the request.

If we want to **continue** with the request, we should log a warning.